src/utils/dom/awaitSelector.js

```diff
@@ -0,0 +1,132 @@
+/*
+Copyright 2019 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+import isFunction from "../isFunction";
+import isNonEmptyArray from "../isNonEmptyArray";
+import Promise from "../Promise";
+import selectNodes from "./selectNodes";
+
+const MUTATION_OBSERVER = "MutationObserver";
+const RAF = "requestAnimationFrame";
+const MUTATION_OBSERVER_CONFIG = { childList: true, subtree: true };
+const VISIBILITY_STATE = "visibilityState";
+const VISIBLE = "visible";
+const DELAY = 100;
+const MAX_POLLING_TIMEOUT = 5000;
+
+function createError(selector) {
```

Comment:
I just noticed this and thought I'd bring up a discussion from a prior PR: https://git.corp.adobe.com/Activation/a-tag/pull/34#discussion_r1725571

I don't know how much the team cares about inconsistencies between function declarations vs function expressions, but if we want to stick to one or the other, we should probably add https://eslint.org/docs/rules/func-style to our eslint rules. I can do that in a separate PR and update any inconsistencies.

Reply:
@Aaronius let's decide and stick to one format, I agree that we should be consistent, I don't mind using func expressions. If we can enforce it via ESLint rules even better.